ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -522,6 +524,62 @@ public void shouldReturn5xxOnStatementSystemError() {
     EasyMock.verify(ksqlEngine);
   }
 
+  @Test
+  public void shouldFailIfReachedActivePersistentQueriesLimit() {
+    // Given:
+    givenKsqlConfigWith(
+        ImmutableMap.of(KsqlConfig.KSQL_ACTIVE_PERSISTENT_QUERY_LIMIT_CONFIG, 3));
+    final String ksqlString = "CREATE STREAM test_explain AS SELECT * FROM test_stream;";
+    givenMockEngine(mockEngine -> {
+      EasyMock.expect(mockEngine.parseStatements(EasyMock.anyString()))
+          .andDelegateTo(realEngine);
+      EasyMock.expect(
+          mockEngine.numberOfPersistentQueries())
+          .andReturn(3L);
+    });
+
+    // When:
+    final KsqlErrorMessage result = makeFailingRequest(
+        ksqlString, Code.BAD_REQUEST);
+
+    // Then:
+    assertThat(result, is(instanceOf(KsqlErrorMessage.class)));
```

Comment:
ermm.... I think the compile is ensuring this!  The type of `results` is.... ?

;)